src/panels/lovelace/cards/hui-active-alerts-card.ts

```diff
@@ -0,0 +1,207 @@
+import { LitElement, html, css } from "lit";
+import { customElement, property, state } from "lit/decorators";
+import type { HomeAssistant } from "../../../types";
+import "../../../components/ha-card";
+import "../../../components/ha-icon";
+
+interface ActiveEntity {
+  entity_id: string;
+  state: string;
+  domain: string;
+  name: string;
+  icon?: string;
+  last_changed?: string;
+}
+
+@customElement("hui-active-alerts-card")
+export class HuiActiveAlertsCard extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @state() private _entities: ActiveEntity[] = [];
+
+  private _config: any;
+
+  private _unsubscribe?: () => void;
+
+  setConfig(config: any) {
+    if (!config) {
+      throw new Error("Invalid configuration");
+    }
+    this._config = {
+      title: config.title || "",
+      entities: config.entities || [],
+    };
+    this._updateEntities();
+  }
+
+  async connectedCallback() {
+    super.connectedCallback();
+    await this._subscribe();
+    this._updateEntities();
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    if (this._unsubscribe) {
+      this._unsubscribe();
+      this._unsubscribe = undefined;
+    }
+  }
+
+  hassChanged(_hass: HomeAssistant) {
+    this._updateEntities();
+  }
+
+  private async _subscribe() {
+    if (!this.hass) return;
+    this._unsubscribe = await this.hass.connection.subscribeEvents(() => {
+      this._updateEntities();
+    }, "state_changed");
+  }
+
+  private _updateEntities() {
+    if (!this.hass) return;
+    const entities: ActiveEntity[] = [];
+    const targetEntities =
+      this._config && this._config.entities && this._config.entities.length > 0
+        ? this._config.entities
```

Comment:
Filter the entities on the correct domain here, instead of in the for loop later.